Sources/HTTP/Responder/HTTPClientProtocolUpgrader.swift

```diff
@@ -118,28 +117,32 @@ private final class HTTPClientUpgradeHandler<Upgrader>: ChannelInboundHandler wh
             return
         }
 
-        // We're trying to remove ourselves from the pipeline but not upgrading, so just pass this on.
-        if seenFirstResponse {
-            ctx.fireChannelRead(data)
-            return
-        }
-
         let responsePart = unwrapInboundIn(data)
-        seenFirstResponse = true
 
         // We should only ever see a request header: by the time the body comes in we should
         // be out of the pipeline. Anything else is an error.
-        guard case .head(let res) = responsePart, res.status == .switchingProtocols, upgrader.isValidUpgradeResponse(res) else {
+        if case .head(let res) = responsePart, res.status == .switchingProtocols, upgrader.isValidUpgradeResponse(res) {
+            self.res = res
+        } else if case .end(_) = responsePart, let res = self.res {
+            upgrading = true
+            
+            removeExtraHandlers(ctx: ctx).then {
+                return self.upgrader.upgrade(ctx: ctx, upgradeResponse: res)
+                }.then { handler in
```

Comment:
small formatting issue here.

Reply:
This seems to be Xcode's preferred formatting, hitting CTRL + I (re-indent) changes nothing

Reply:
Fixed it 😄